graphdatascience/tests/integration/test_database_ops.py

```diff
@@ -49,6 +49,27 @@ def test_run_query_with_db(runner: Neo4jQueryRunner) -> None:
     runner.run_query("DROP DATABASE $dbName", {"dbName": MY_DB_NAME})
 
 
+@pytest.mark.skip_on_aura
+def test_initialize_with_db(runner: Neo4jQueryRunner) -> None:
+    runner.run_query("CREATE (a: Node)")
+
+    default_db_count = runner.run_query("MATCH (n: Node) RETURN COUNT(n) AS c")["c"][0]
+    assert default_db_count == 1
+
+    MY_DB_NAME = "my-db"
+    runner.run_query("CREATE DATABASE $dbName", {"dbName": MY_DB_NAME})
+
+    gds_with_specified_db = GraphDataScience(URI, AUTH, database=MY_DB_NAME)
+
+    specified_db_count = gds_with_specified_db.run_cypher("MATCH (n: Node) RETURN COUNT(n) AS c", database=MY_DB_NAME)[
+        "c"
+    ][0]
+    assert specified_db_count == 0
+
+    runner.run_query("MATCH (n) DETACH DELETE n")
+    runner.run_query("DROP DATABASE $dbName", {"dbName": MY_DB_NAME})
```

Comment:
if the assert fails, we wont do the cleanup.
Shouldnt we use a try + finally here?

Reply:
You're right. And we should have some global way of doing this for all tests too. That's for another time though. I'll adopt your suggestion

Reply:
done